elide-core/src/main/java/com/yahoo/elide/Elide.java

```diff
@@ -309,6 +348,12 @@ protected ElideResponse handleRequest(boolean isReadOnly, User user,
                 requestScope.getPermissionExecutor().printCheckStats();
             }
 
+            //call complete query here
+            if (requestScope.getElideSettings() != null) {
+                QueryLogger ql = requestScope.getElideSettings().getQueryLogger();
+                ql.completeQuery(UUID.fromString(requestScope.getRequestId()), response);
```

Comment:
This probably needs to be in a finally block to avoid leaks.

Reply:
Umm sorry but, I cannot see where the leaks might come from. Can you elaborate a bit more please?